webpack/scenes/ContentViews/Table/ContentViewsTable.js

```diff
@@ -0,0 +1,132 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+} from '@patternfly/react-table';
+import PropTypes from 'prop-types';
+import { translate as __ } from 'foremanReact/common/I18n';
+
+import Loading from './Loading';
+import EmptyStateMessage from '../components/EmptyStateMessage';
+import tableDataGenerator from './tableDataGenerator';
+import './ContentViewsTable.scss';
+
+const ContentViewTable = ({
+  loadContentViewDetails, detailsMap, results, loading,
+}) => {
+  const [table, setTable] = useState({ rows: [], columns: [] });
```

Comment:
@johnpmitsch (Documenting/expanding our IRC discussion here)

I think that since `expandedColumnMap` only stores data about what rows are expanded, the right place for this is in internal component state.

For the `table` rows and columns, I can see the argument for moving to Redux.  The task is to take the raw API response and format it in the way that PF4 requires.  The question is where should this "data normalization" be done?

There are a few possible approaches:
1. Use the `react-redux` hook `useSelector` to read the API response.  Format the data in the component using `tableDataGenerator` like you're already doing.  (May require some refactoring of `tableDataGenerator`, but otherwise not too much change
2. Write a custom hook, `useForemanTableData`, that does the same thing as the above.  This way it could be reused across several tables.
3. Have the Redux reducer format the API response and just store the pre-formatted data in Redux.  Then you could use `useSelector` in the component but not worry about data formatting there.
4. Write a new Redux middleware that looks for the Content View SUCCESS action, formats the data, and dispatches a new action which stores the formatted data in Redux.  This keeps business logic out of the reducer, which really should be only for transforming actions into new state.

Any of these will avoid using `setState` for the table data.  I'd be perfectly happy with #1 for now, and either #2 or #4 eventually.

Reply:
Which makes me think, @laviro is there a way for plugins to add Redux middleware to Foreman?  I think we'd probably have to build that.

Reply:
I think that the more we divide the markup code with the logic, is easier to maintain the component and if one day we will want to re-use the component elsewhere it will be much easier too.

as for 1), I think our goal is to pass the props as close as the PF component expects,
and make all of the logic in selectors, actions, reducers, constants.

2) sounds like a great idea!

3) really soon in https://github.com/theforeman/foreman/pull/7337 we can use callbacks in the API middleware, but as for now, it's totally fine to listen to the SUCCESS case in your reducer.

4) I think no. 3) solve this issue.

> is there a way for plugins to add Redux middleware to Foreman?

lol, that's right.. I actually think they can't at the moment, but maybe it's better in the case of middlewares to live in foreman-core as they are affecting every dispatched action

Reply:
The `handleSuccess` callback looks like it may be a great replacement for 4!